cdc/sorter/leveldb/message/task.go

```diff
@@ -27,11 +28,20 @@ type Task struct {
 	UID     uint32
 	TableID uint64
 
+	// Input unsorted event for writers.
+	// Sorter.AddEntry -> writer.
+	InputEvent *model.PolymorphicEvent
+	// Latest resolved ts / commit ts for readers.
+	// writer -> reader
+	ReadTs ReadTs
```

Comment:
It makes the logic complex and unreadable to add so many fields in one struct, some fields are always nil in some cases.
And the `Task` is a part of `Message`,  The size will be larger and larger, maybe we should find a better way to extend the `Task` or `Message`.

Reply:
It's good point! Ideally tasks should be an enum type in rust, so tasks can be separated and more readable. Unfortunately, golang does not support it.

> And the Task is a part of Message, The size will be larger and larger, maybe we should find a better way to extend the Task or Message.

Maybe we should make Task be a pointer in Message? Though it makes memory allocation, not sure if it helps performance.

Reply:
Also see, notes in #4631